File: Data Structures & Algorithms/permutations/submission-0.java
class Solution {
    List<List<Integer>> result = new ArrayList<>();
    boolean[] used ;
    public List<List<Integer>> permute(int[] nums) {
        used = new boolean[nums.length];
        backtrack(nums,new ArrayList<>());
        return result;
    }
    public void backtrack(int[] nums,List<Integer> path){
        
        if(path.size()==nums.length){
            result.add(new ArrayList<>(path));
            return;
        }
        for(int i=0;i<nums.length;i++)
        {
            if(used[i]==true)
            {
                continue;
            }
            used[i]=true;
            path.add(nums[i]);
            backtrack(nums,path);
            path.remove(path.size()-1);
            used[i]=false;
        }
        
    }
}

File: Data Structures & Algorithms/concatenation-of-array/submission-0.cpp
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        int size = nums.size();
        vector<int> result (2*size);
        for(int i=0;i<2*size;i++){
            result[i] = nums[i%size];
        }
        return result;
    }
};